Micronaut Maven Project Generation Fails: 'artifactId' is missing in pom.xml

### Issue description

When generating a new Micronaut project with Maven, the pom.xml is not being generated correctly. Upon running mvn compile (or any Maven command), the following error occurs:
```
[INFO] Scanning for projects...
[ERROR] Some problems were encountered while processing the POMs:
[FATAL] 'artifactId' is missing. @ line 3, column 109

[ERROR] The project io.micronaut.platform:[unknown-artifact-id]:4.9.1 
($home/demo/pom.xml) has 1 error
[ERROR] 'artifactId' is missing. @ line 3, column 109

```

--- a/projectgen-core/src/main/java/io/micronaut/projectgen/core/options/GenericOptions.java
+++ b/projectgen-core/src/main/java/io/micronaut/projectgen/core/options/GenericOptions.java
@@ -79,6 +79,22 @@ public record GenericOptions(@NonNull String name,
     }
 
     @Override
+    public String artifact() {
+        if (StringUtils.isEmpty(this.artifact)) {
+            return name();
+        }
+        return this.artifact;
+    }
+
+    @Override
+    public String group() {
+        if (StringUtils.isEmpty(this.group)) {
+            return packageName();
+        }
+        return this.group;
+    }
+
+    @Override
     public List<String> features() {
         return CollectionUtils.isEmpty(this.features) ? Collections.emptyList() : this.features;
     }

--- a/test-suite-projectgen-micronaut/src/main/java/io/micronaut/projectgen/micronaut/ApplicationTypeFeature.java
+++ b/test-suite-projectgen-micronaut/src/main/java/io/micronaut/projectgen/micronaut/ApplicationTypeFeature.java
@@ -18,12 +18,15 @@ package io.micronaut.projectgen.micronaut;
 import io.micronaut.projectgen.core.buildtools.MavenCentral;
 import io.micronaut.projectgen.core.buildtools.Repository;
 import io.micronaut.projectgen.core.buildtools.RequiresRepository;
+import io.micronaut.projectgen.core.buildtools.dependencies.Dependency;
 import io.micronaut.projectgen.core.feature.ConfigurationFeature;
 import io.micronaut.projectgen.core.feature.DefaultFeature;
 import io.micronaut.projectgen.core.feature.FeatureContext;
 import io.micronaut.projectgen.core.feature.LoggingFeature;
 import io.micronaut.projectgen.core.feature.config.Properties;
 import io.micronaut.projectgen.core.feature.gitignore.GitIgnore;
+import io.micronaut.projectgen.core.generator.GeneratorContext;
+import io.micronaut.projectgen.core.options.Options;
 import io.micronaut.projectgen.core.options.TestFramework;
 import io.micronaut.projectgen.micronaut.features.cli.MicronautCli;
 import io.micronaut.projectgen.micronaut.features.logging.Logback;
@@ -62,6 +65,16 @@ public abstract class ApplicationTypeFeature implements DefaultFeature, Requires
         } else if (featureContext.getOptions().testFramework() == TestFramework.SPOCK) {
             featureContext.addFeatureIfNotPresent(MicronautTestSpock.class, micronautTestSpock);
         }
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        Options options = generatorContext.getOptions();
+        generatorContext.getRootModule().moduleAttributes().setCoordinate(Dependency.builder()
+            .groupId(options.group())
+            .artifactId(options.artifact())
+            .version(options.version())
+            .build());
     }
 
     @Override

--- a/test-suite-projectgen-micronaut/src/test/java/io/micronaut/projectgen/micronaut/MicronautApplicationGenerationTest.java
+++ b/test-suite-projectgen-micronaut/src/test/java/io/micronaut/projectgen/micronaut/MicronautApplicationGenerationTest.java
@@ -48,8 +48,11 @@ class MicronautApplicationGenerationTest {
         expected.addAll(EXPECT_FILES_FOR_BOTH);
         assertEquals(expected, project.keySet());
         String pomXml = project.get("pom.xml");
-        System.out.println(pomXml);
         BuildTestVerifier verifier = BuildTestVerifier.of(pomXml, BuildTool.MAVEN, options.language(), options.testFramework());
+        assertTrue(verifier.hasParentPom("io.micronaut.platform", "micronaut-parent"), pomXml);
+        assertTrue(pomXml.contains("<groupId>com.example</groupId>"), pomXml);
+        assertTrue(pomXml.contains("<artifactId>demo</artifactId>"), pomXml);
+        assertTrue(pomXml.contains("<version>0.1</version>"), pomXml);
         assertEquals("netty", verifier.getProperty("micronaut.runtime"));
         assertEquals("jar", verifier.getProperty("packaging"));
         assertEquals("21", verifier.getProperty("jdk.version"));
@@ -104,6 +107,7 @@ class MicronautApplicationGenerationTest {
         return OptionsFixture.defaultNoBuildTool()
             .template(ApplicationType.DEFAULT.toString())
             .name("demo")
+            .version("0.1")
             .packageName("com.example")
             .java(JdkVersion.JDK_21)
             .gradleDsl(GradleDsl.KOTLIN)

